### PR TITLE
Hotfix ownerreference

### PR DESCRIPTION
**Pull request by YiscahLevySilas1**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,85 +3,101 @@ module github.com/kubescape/rbac-utils
 go 1.18
```
Comment (dwertent) [inline go.mod]: Please upgrade to `1.20`

**Review by dwertent — CHANGES_REQUESTED** [review]

File: rbacutils/rbacprocessor.go [hunk 2]
```diff
@@ -65,12 +64,15 @@ func InitSAID2WLIDmap(k8sAPI *k8sinterface.KubernetesApi, clusterName string) (m
...
 	}
 	for _, wl := range allworkloads {
-		if ref, err := wl.GetOwnerReferences(); len(ref) == 0 && err == nil {
-			saID := fmt.Sprintf("/%s/%s/%s/%s", serviceaccountversion, wl.GetNamespace(), serviceaccountkind, wl.GetServiceAccountName())
+		if !k8sinterface.WorkloadHasParent(wl) {
+			connectedSA := wl.GetServiceAccountName()
+			if connectedSA == "" {
+				connectedSA = "default"
+			}
+			saID := fmt.Sprintf("/%s/%s/%s/%s", serviceaccountversion, wl.GetNamespace(), serviceaccountkind, connectedSA)
```
Comment (dwertent) [inline rbacutils/rbacprocessor.go]: Use `strings.Join` instead.

File: rbacutils/rbacprocessor.go [hunk 3]
```diff
@@ -82,7 +84,7 @@ func InitSAID2WLIDmap(k8sAPI *k8sinterface.KubernetesApi, clusterName string) (m
 	return saID2WLIDmap, nil
 }
 
-//TODO - DEPRECATE sa2WLIDmap
+// TODO - DEPRECATE sa2WLIDmap
```
Comment (dwertent) [inline rbacutils/rbacprocessor.go]: Could this be removed?

File: rbacutils/rbacprocessor.go [hunk 4]
```diff
@@ -220,7 +221,7 @@ func ExistsSubject(list []Subject, subjectName string) (int, bool) {
 //  =========================== rbac table ======================
 
 // DEPRECATED
-//InitRbacTable -
+// InitRbacTable -
 func InitRbacTable(clustername string, clusterRoles *rbac.ClusterRoleList, roles *rbac.RoleList, clusterRoleBindings *rbac.ClusterRoleBindingList, roleBindings *rbac.RoleBindingList) *[]RbacTable {
```
Comment (dwertent) [inline rbacutils/rbacprocessor.go]: Could this be removed?

**Review by dwertent — CHANGES_REQUESTED** [review]

File: rbacutils/rbacprocessor_test.go [hunk 5]
```diff
@@ -1,33 +1 @@
 package rbacutils
-
```
Comment (dwertent) [inline rbacutils/rbacprocessor_test.go]: remove the file if there are no tests

File: rbacutils/rbacprocessor.go (hunk 3 above)
Comment (YiscahLevySilas1) [reply rbacutils/rbacprocessor.go]: yes, we'll need to deprecate it in ks and BE as well

File: rbacutils/rbacprocessor.go [hunk 6]
```diff
@@ -65,12 +63,15 @@ func InitSAID2WLIDmap(k8sAPI *k8sinterface.KubernetesApi, clusterName string) (m
...
 	}
 	for _, wl := range allworkloads {
-		if ref, err := wl.GetOwnerReferences(); len(ref) == 0 && err == nil {
-			saID := fmt.Sprintf("/%s/%s/%s/%s", serviceaccountversion, wl.GetNamespace(), serviceaccountkind, wl.GetServiceAccountName())
+		if !k8sinterface.WorkloadHasParent(wl) {
+			connectedSA := wl.GetServiceAccountName()
+			if connectedSA == "" {
+				connectedSA = "default"
+			}
+			saID := "/" + strings.Join([]string{serviceaccountversion, wl.GetNamespace(), serviceaccountkind, connectedSA}, "/")
```
Comment (dwertent) [inline rbacutils/rbacprocessor.go:74]: Do we parse the ID anywhere? If not, we don't need to add the "/" prefix.

File: go.mod [hunk 7]
```diff
@@ -1,86 +1,103 @@
 module github.com/kubescape/rbac-utils
 
-go 1.18
+go 1.20
```
Comment (dwertent) [inline go.mod:3]: You will need to update the github workflows as well :)

**Review by dwertent — CHANGES_REQUESTED** [review]

File: rbacutils/rbacprocessor.go (hunk 6 above)
Comment (YiscahLevySilas1) [reply rbacutils/rbacprocessor.go:74]: In the UI, I don't want to break anything there

**Pull request closed without merging**